pkg/api/componentreadiness/component_report.go

```diff
@@ -504,6 +545,40 @@ func (c *componentReportGenerator) getBaseQueryStatus(ctx context.Context,
 	return componentReportTestStatus.BaseStatus, nil
 }
 
+func (c *componentReportGenerator) getFallbackBaseQueryStatus(ctx context.Context, allJobVariants crtype.JobVariants, release string, start, end time.Time) []error {
+	baseQuery, baseGrouping, baseParams := c.getCommonTestStatusQuery(allJobVariants, false, true)
+	generator := fallbackTestQueryReleasesGenerator{
+		client: c.client,
+		cacheOption: cache.RequestOptions{
+			ForceRefresh: c.cacheOption.ForceRefresh,
+			// increase the time that fallback queries are cached for
+			// could formalize as input flag
+			CRTimeRoundingFactor: fallbackQueryTimeRoundingOverride,
+		},
+		commonQuery:     baseQuery,
+		groupByQuery:    baseGrouping,
+		BaseRelease:     release,
+		BaseStart:       start,
+		BaseEnd:         end,
+		queryParameters: baseParams,
+	}
+
+	// default to main cache
+	// but if defined will use persistent cache
+	fcache := c.client.Cache
+	if c.client.PersistentCache != nil {
+		fcache = c.client.PersistentCache
+	}
```

Comment:
This feels a little off in ways that will be confusing to explain to someone. The passing of a cache, but there's another cache embedded on the bigquery client that can be used for other purposes but also proxy off the original cache. Could we come up with a more cohesive mechanism for this module?

Passing in the bigquery cache separately would be one step clearer in my mind, its unexpected it would be hanging off the client as the two then depend on each other.

But ideally a single cache that could handle all the logic would be really nice. It could know to make use or the bigquery cache or not based on the size of the data and the duration requested to cache for. Though I don't know how we could tell it not to fallback to the bigquery cache for most regular pre-existing keys.

Reply:
If there is support for the change we could move everything to a single cache that would only attempt to persist items that have a retention time longer than X and/or byte length > Y, etc.

Reply:
That would be excellent and really nicely encapsulate the complexity.

Reply:
Had to modify the Get url to include duration but added logic in the bigquery cache to not persist data below a min duration and collapsed down to a single cache from the component_report perspective.